src/OpenLoco/Paint/PaintMiscEntity.cpp

```diff
@@ -0,0 +1,323 @@
+    }
+
+    static void paintWindowCurrencyEntity(PaintSession& session, MoneyEffect* moneyEffect)
+    {
+        Gfx::drawpixelinfo_t* dpi = session.getContext();
+        int8_t* wiggleYOffsets = reinterpret_cast<int8_t*>(0x4FAAC8);
+        if (dpi->zoom_level > 1)
+        {
+            return;
+        }
+        const string_id stringId = moneyEffect->amount >= 0 ? StringIds::format_currency_income_in_company_colour : StringIds::format_currency_expense_in_company_colour_negative;
+        uint32_t currencyAmount = abs(moneyEffect->amount);
+        int8_t* yOffsets = &wiggleYOffsets[moneyEffect->wiggle];
+        uint16_t companyColour = CompanyManager::getCompanyColour(moneyEffect->var_2E);
+
+        session.addToStringPlotList(currencyAmount, stringId, moneyEffect->y, moneyEffect->z, yOffsets, moneyEffect->offsetX, companyColour);
+    }
+
+    static void paintVehicleCrashParticleEntity(PaintSession& session, VehicleCrashParticle* particle)
+    {
+        Gfx::drawpixelinfo_t* dpi = session.getContext();
+        if (dpi->zoom_level != 0)
+        {
+            return;
+        }
+
+        loco_global<int32_t[5], 0x4FAAB4> vehicle_particle_base_sprites;
+        uint32_t imageId = vehicle_particle_base_sprites[particle->crashedSpriteBase] + particle->frame / 256;
+        imageId = Gfx::recolour2(imageId, particle->colourScheme);
+
+        session.addToPlotListAsParent(imageId, { 0, 0, particle->z }, { 1, 1, 0 });
+    }
+
+    static void paintExplosionCloudEntity(PaintSession& session, ExplosionCloud* particle)
+    {
+        Gfx::drawpixelinfo_t* dpi = session.getContext();
+        if (dpi->zoom_level > 2)
+        {
+            return;
+        }
+
+        static const std::array<uint32_t, 18> explosionCloudImageIds = {
+            ImageIds::explosion_cloud_00,
+            ImageIds::explosion_cloud_01,
+            ImageIds::explosion_cloud_02,
+            ImageIds::explosion_cloud_03,
+            ImageIds::explosion_cloud_04,
+            ImageIds::explosion_cloud_05,
+            ImageIds::explosion_cloud_06,
+            ImageIds::explosion_cloud_07,
+            ImageIds::explosion_cloud_08,
+            ImageIds::explosion_cloud_09,
+            ImageIds::explosion_cloud_10,
+            ImageIds::explosion_cloud_11,
+            ImageIds::explosion_cloud_12,
+            ImageIds::explosion_cloud_13,
+            ImageIds::explosion_cloud_14,
+            ImageIds::explosion_cloud_15,
+            ImageIds::explosion_cloud_16,
+            ImageIds::explosion_cloud_17
+        };
+
+        assert(static_cast<size_t>(particle->frame / 256) < explosionCloudImageIds.size());
+        uint32_t imageId = explosionCloudImageIds.at(particle->frame / 256);
+        session.addToPlotListAsParent(imageId, { 0, 0, particle->z }, { 1, 1, 0 });
+    }
+
+    static void paintSparkEntity(PaintSession& session, Spark* particle)
```

Comment:
This seems to be drawing water splashes, not sparks.

Reply:
renamed to splash now.